### DISCUSSION TITLE
how do I make other gates

### DISCUSSION THREAD

**Discussion by yaay1010** [SebLague/Digital-Logic-Sim#54] [General]
### _**Hi, I want to make gates like XOR, OR, NAND, and others. Can someone record them making one? Thanks in advance :)**_

**Comment by Fisch37** [comment]
Uhhh, so NAND is short for NOT-AND. That should tell you how to do that.
And if I remember correctly the OR gate is a NAND Gate with two inverted inputs.

**Comment by Subroutine7901** [comment]
i don't have a screen shot but in text i'll try to explain

or:

`input -- not\`
`=========== nand -- output`
`input -- not/`

xor:

`input\------/nand\`
`======nand<====== nand--output`
`input/------\nand/`

the =='s are placeholders

i'm sorry if this is confusing

**Reply by Subroutine7901** [reply]
also if a gates name starts with an n then it's like the example Leonix1 gave earlier and xnor is like the "n" in it's name is just at the start

**Comment by ghost** [comment]
NAND:
![grafik](https://user-images.githubusercontent.com/62117018/150872978-e8fc8c01-3bf3-4854-a7ef-1c939cc2d43d.png)

**Truth Table**
> A | B | Q
> 0 | 0 | 1
> 0 | 1 | 1
> 1 | 0 | 1
> 1 | 1 | 0

--------------------------------------------------------------------------------

OR:
![grafik](https://user-images.githubusercontent.com/62117018/150873155-c2bc3a9c-146d-4c66-a13d-aef8cfa0d2d1.png)

**Truth Table**
> A | B | Q
> 0 | 0 | 0
> 0 | 1 | 1
> 1 | 0 | 1
> 1 | 1 | 1

--------------------------------------------------------------------------------

NOR:
![grafik](https://user-images.githubusercontent.com/62117018/150873265-dc82a4e0-d99d-41dd-8306-e37bffc22780.png)

**Truth Table**
> A | B | Q
> 0 | 0 | 1
> 0 | 1 | 0
> 1 | 0 | 0
> 1 | 1 | 0

--------------------------------------------------------------------------------
XOR:
![grafik](https://user-images.githubusercontent.com/62117018/150873433-ccaf6af8-44aa-46e8-bb6a-dc7e5bf6640d.png)

**Truth Table**
> A | B | Q
> 0 | 0 | 0
> 0 | 1 | 1
> 1 | 0 | 1
> 1 | 1 | 0

--------------------------------------------------------------------------------
XNOR;
![grafik](https://user-images.githubusercontent.com/62117018/150873570-3f31b2de-9918-476c-9289-4d1350a8e9eb.png)

**Truth Table**
> A | B | Q
> 0 | 0 | 1
> 0 | 1 | 0
> 1 | 0 | 0
> 1 | 1 | 1

--------------------------------------------------------------------------------

**Comment by gegentan** [comment]
When you make a NOT gate chip, how do you make only one of the 1-bit inputs appear on the chip inputs?

<img width="1744" height="625" alt="image" src="https://github.com/user-attachments/assets/2fe1131d-49b5-4e6d-9581-10420f522292" />
<img width="1166" height="544" alt="image" src="https://github.com/user-attachments/assets/87d077d3-a54d-4c45-879d-57837936d8d4" />

(The red cross is the input on the chip I want gone)

**Reply by Noobest1001** [reply]
You make it have only one input and attach the input to both inputs of the NAND gate
